pyanaconda/core/kickstart/addon.py

```diff
@@ -0,0 +1,182 @@
+#
+# Support for %addon sections.
+#
+# Copyright (C) 2019  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+from abc import ABCMeta, abstractmethod
+from types import SimpleNamespace
+
+from pykickstart.errors import KickstartParseError
+from pykickstart.ko import KickstartObject
+from pykickstart.sections import Section
+
+from pyanaconda.anaconda_loggers import get_module_logger
+from pyanaconda.core.i18n import _
+
+log = get_module_logger(__name__)
+
+__all__ = ["AddonData", "AddonSection", "AddonRegistry"]
+
+
+class AddonData(KickstartObject, metaclass=ABCMeta):
+    """Kickstart data for addon.
+
+    This is a common parent class for loading and storing 3rd
+    party data to kickstart. It is instantiated by kickstart
+    parser and stored as ksdata.addons.<name> to be used in
+    the user interfaces.
+
+    The following methods have to be implemented. The method
+    handle_header should handle arguments of the section. The
+    method handle_line is called for every line of the section.
+    The method __str__ should return a string with a kickstart
+    representation of the %addon section.
```

Comment:
Could you please format this as a list instead of sentence?

```
The following methods...:
* handle_header - handle arguments of the section
* handle_linte - ...
...
```

Reply:
Fixed.